src/applications/app.helpers.js

```diff
@@ -36,15 +36,53 @@ export function isntLoaded(app) {
 
 export function shouldBeActive(app) {
   try {
-    return app.activeWhen(window.location);
+    if (Array.isArray(app.activeWhen))
+      return app.activeWhen.some(conditionForAppActivationMet);
+    return conditionForAppActivationMet(app.activeWhen);
   } catch (err) {
     handleAppError(err, app, SKIP_BECAUSE_BROKEN);
   }
+
+  function conditionForAppActivationMet(pathPrefixOrActivityFn) {
+    const { location } = window;
+    if (typeof pathPrefixOrActivityFn === "string") {
+      const hashRouting = pathPrefixOrActivityFn.startsWith("/#/");
+      const [
+        dynamicIndexes,
+        pathPrefixWithoutDynamic,
+      ] = pathPrefixOrActivityFn.split("/").reduce(
```

Comment:
I think this implementation could be simpler if we  if we created a regular expression from the string, and then implemented the activity function to be `regex.test()`.

Here's an implementation that might be a bit simpler. I find it to be a bit easier to follow, but ultimately the difference between the implementations might fall into world of preference

```js
function stringToActivityFn(path) {
  const regex = toRegex(path);
  return function (location) {
    const route = location.href.replace(location.origin, '');
    return regex.test(route)
  }
}

function toRegex(path) {
  let lastIndex = 0, inDynamic = false, regexStr = '^';

  for (let i = 0; i < path.length; i++) {
    const char = path[i];
    if (inDynamic && char === '/' || !inDynamic && char === ':') {
      appendToRegex(i);
    }
  }

  appendToRegex(path.length);

  return new RegExp(regexStr, 'i');

  function appendToRegex(index) {
    regexStr += inDynamic ? '.+?' : escapeStrRegex(path.slice(lastIndex, index));
    inDynamic = !inDynamic;
    lastIndex = index;
  }
}

function escapeStrRegex(str) {
  // borrowed from https://github.com/sindresorhus/escape-string-regexp/blob/master/index.js
  return str.replace(/[|\\{}()[\]^$+*?.]/g, '\\$&')
}
```

Reply:
I agree that the `stringToActivityFn` is now 200% more simpler and easier to read, but since I lack regex skills, `toRegex` is hard to follow, so i'm here trying to learn what I can from this implementation haha

Basically you want to transform the `path` the user can input to a regex which can match the location, right?
Sounds like a plan, because if you want to give users the possibility of declaring routes like: `/onlyLetters/:[a-Z]*/forSomeReason` you just need to enhance this "append to regex" function! 🥇 
2 question though: 
- this implementation isn't catching
`toRegex('/route/:companyId/subroute').test('/route//subroute')`, when no dynamic value is on location (which is a case for my company, more like a bug, but exists in some places kkk)
Is it intended? 
- What is the `/?` n `.+/?` used for? optional `/` in dynamic value?

I will change my implementation to this one! I know some things are on the realm of preferences, but I think this implementation has the flexibility to evolve to the things I said, which my can't give nothing... I do like reduce over iterating over each char though hehe

Reply:
> Basically you want to transform the path the user can input to a regex which can match the location, right?

Yes

> this implementation isn't catching
toRegex('/route/:companyId/subroute').test('/route//subroute'), when no dynamic value is on location (which is a case for my company, more like a bug, but exists in some places kkk)
Is it intended?

In my opinion, this should return `false`. When a route parameter is provided, I think the normal thing is to require it to be present. If the user wants to make it optional, they could provide two strings:

```js
['/route/:companyId/subroute', '/route/subroute']
```

> What is the /? n .+/? used for? optional / in dynamic value?

`.+` means "at least one character of any kind". The `+` operator is ["greedy"](https://stackoverflow.com/questions/2301285/what-do-lazy-and-greedy-mean-in-the-context-of-regular-expressions), but we only want it to go up to the next `/` instead of for the remainder of the string. This is something worth adding tests for The `/?` indicates "look for a `/` as the end of the dynamic param, but if it's not there, that's okay." The reason the trailing slash would not be there is if the string ends in a dynamic route `/route/:thing`

Reply:
That regex looks great. From a maintenance perspective I'd like to see a handful of tests around the `excapeStrRegex` method. That way we can test as many scenarios as we need, and if it fails to handle something a bug report can include a failing test.